elm/base.py

```diff
@@ -46,7 +46,10 @@ class ApiBase(ABC):
 
     TOKENIZER_ALIASES = {'gpt-35-turbo': 'gpt-3.5-turbo',
                          'gpt-4-32k': 'gpt-4-32k-0314',
-                         'llmev-gpt-4-32k': 'gpt-4-32k-0314'
+                         'llmev-gpt-4-32k': 'gpt-4-32k-0314',
+                         'wetosa-gpt-4': 'gpt-4',
+                         'wetosa-gpt-4-standard': 'gpt-4',
+                         'wetosa-gpt-4o': 'gpt-4o',
                          }
```

Comment:
Can we fix this up somehow so the tokenizer uses "gpt-4" if it cant find anything in the tokenizer alias. I feel like we had this at one point and i dont know this is just annoying lol.

Reply:
What we've had up until this point is to use the actual model name itself if it doesn't exist in the alias dictionary:

https://github.com/NREL/elm/blob/2e4f2685b7ec598077a0b81ddb5298446fe94589/elm/base.py#L364

Just making sure you want to update this behavior to always default to gpt-4?

Reply:
What do you think about this:

Put TOKENIZER_PATTERNS in the class attrs:
```
    TOKENIZER_PATTERNS = ('gpt-4o', 'gpt-4-32k', 'gpt-4')
    """Order-prioritized list of model sub-strings to look for in model name
    to send to tokenizer. As an alternative to alias lookup, this will use the
    tokenizer pattern if found in the model string"""
```

Refactor method to look for patterns:
```
    @classmethod
    def count_tokens(cls, text, model, fallback_model='gpt-4'):
        """Return the number of tokens in a string.

        Parameters
        ----------
        text : str
            Text string to get number of tokens for
        model : str
            specification of OpenAI model to use (e.g., "gpt-3.5-turbo")
        fallback_model : str
            Model to be used for tokenizer if input model can't be found in
            TOKENIZER_ALIASES and doesn't have any easily noticeable patterns.

        Returns
        -------
        n : int
            Number of tokens in text
        """

        if model in cls.TOKENIZER_ALIASES:
            token_model = cls.TOKENIZER_ALIASES[model]
        else:
            token_model = fallback_model
            for pattern in cls.TOKENIZER_PATTERNS:
                if pattern in model:
                    token_model = pattern
                    break

        encoding = tiktoken.encoding_for_model(token_model)

        return len(encoding.encode(text))

Reply:
Looks good to me! Added it in

Reply:
thank you!